src/server-code/admin-server.js

```diff
@@ -141,23 +140,20 @@ export default function runAdminServer(serverFalcorModel) {
 
   app.post('/upload', upload.single('image'), (req, res) => {
     const filePath = req.file.path;
-    const deleteTmpFile = () => {
-      fs.unlink(filePath, (err) => {
-        if (err) {
-          console.error(err); // eslint-disable-line no-console
-        }
-      });
-    };
+
     if (isDevelopment) {
       /**
        * As we are in dev-mode, we don't actually want to upload to s3.
        * You can either compile with production mode or remove this
        * temporarily if extra s3 tests are needed at some point.
        */
-      setTimeout(() => {
-        res.status(200).send('success test_url');
-      }, 2000);
-      deleteTmpFile();
+      compressJPEG(filePath).then(() => {
+        setTimeout(() => {
+          res.status(200).send('success on test url');
+        }, 2000);
+        // eslint-disable-next-line no-console
+        deleteFile(filePath).catch(error => console.log(error));
```

Comment:
I like the catches btw, very important and reminds me of one of the many things on my to-do list to properly handle errors <.<